pkg/skaffold/build/jib/jib_sync.go

```diff
@@ -0,0 +1,213 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jib
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"time"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/filemon"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+	"github.com/pkg/errors"
+)
+
+var syncLists = make(map[string]SyncMap)
+
+type SyncMap struct {
+	Direct    []SyncEntry `json:"direct"`
+	Generated []SyncEntry `json:"generated"`
+}
+
+type SyncEntry struct {
+	Src  string `json:"src"`
+	Dest string `json:"dest"`
+
+	filetime time.Time
+}
+
+func InitSync(ctx context.Context, workspace string, artifact *latest.JibArtifact) error {
+	syncMap, err := getSyncMap(ctx, workspace, artifact)
+	if (err != nil) {
+		return err
+	}
+	syncLists[artifact.Project] = *syncMap
+	return nil
+}
+
+// returns toCopy, toDelete, error
+func GetSyncDiff(ctx context.Context, workspace string, artifact *latest.JibArtifact, e filemon.Events) (map[string][]string, map[string][]string, error) {
+
+	// if anything that was modified was a buildfile, do NOT sync, do a rebuild
+	buildFiles := GetBuildDefinitions(artifact)
+	for _, f := range e.Modified {
+		if !filepath.IsAbs(f) {
+			if ff, err := filepath.Abs(f); err != nil{
+				return nil, nil, err
+			} else {
+				f = ff
+			}
+		}
+		for _, bf := range buildFiles {
+			if f == bf {
+				return nil, nil, nil
```

Comment:
Considering the method name, I think it's worth adding a comment `nil, nil` means a full rebuild.